rr_common/src/mapper/mapper.cpp

```diff
@@ -13,12 +13,26 @@
 using cloud_t = pcl::PointCloud<pcl::PointXYZ>;
 using cloud_ptr_t = pcl::PointCloud<pcl::PointXYZ>::Ptr;
 
-std::map<std::string, sensor_msgs::PointCloud2ConstPtr> partials;
-bool needsUpdating = false;
+std::map<std::string, std::queue<sensor_msgs::PointCloud2ConstPtr>> queues;
+bool hasChanged = false;
+
+bool shouldUpdate() {
+    if (!hasChanged) {
+        return false;
+    }
+
+    for (const auto& entry : queues) {
+        if (entry.second.empty()) {
```

Comment:
If I am not mistaken, this will only update everything if all of the topics had new data in the queue. It is possible that we are subscribed to a lot of topics but only a few are updated that should still be merged. Is that right? @ebretl

Reply:
That was my understanding of the new desired behavior, could we get clarification on that @ebretl ?

Reply:
We want to publish outputs that synchronize data from multiple inputs. As such, I think Dan's implementation is fine. To address the issue Brian raised, a good additional feature would be to require new data only on topics for which we have ever received a message, or for which we have received data in the past X seconds.